File: task_equivalents/llm_translations/gemini/optimized/python/data_visualization/02_distribution_boxplot.py
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

college = pd.read_csv('data/csv/College.csv', index_col=0)

college['Elite'] = np.where(college['Top10perc'] > 50, 'Yes', 'No')

# First plot: Boxplot (equivalent to plot(factor, numeric) in R)
fig1, ax1 = plt.subplots()
college.boxplot(column='Outstate', by='Elite', ax=ax1, grid=False)
fig1.suptitle('')
ax1.set_title('')
ax1.set_xlabel('Elite')
ax1.set_ylabel('Out-of-state tuition')

# Second plot: 2x2 grid of histograms
fig2, axes = plt.subplots(2, 2, figsize=(10, 8))

axes[0, 0].hist(college['Apps'], bins=50, edgecolor='black')
axes[0, 0].set_xlabel('new applications')

axes[0, 1].hist(college['Enroll'], bins=45, edgecolor='black')
axes[0, 1].set_xlabel('new enrollment')

axes[1, 0].hist(college['Expend'], bins=30, edgecolor='black')
axes[1, 0].set_xlabel('Instructional expenditure per student')

axes[1, 1].hist(college['Outstate'], edgecolor='black')
axes[1, 1].set_xlabel('Out-of-state tuition')

plt.tight_layout()
plt.show()

File: task_equivalents/llm_translations/llama/base/r/data_visualization/02_distribution_boxplot.R
# Load necessary libraries
library(readr)
library(dplyr)
library(ggplot2)

# Read the CSV file
college <- read_csv('data/csv/College.csv')

# Rename the first column to 'School' and set it as the row names
college <- college %>% 
  rename(School = 1) %>% 
  tibble::column_to_rownames('School')

# Convert 'Private' to a factor
college$Private <- as.factor(college$Private)

# Create a new column 'Elite'
college$Elite <- ifelse(college$Top10perc > 50, 'Yes', 'No')
college$Elite <- as.factor(college$Elite)

# Boxplot
ggplot(college, aes(x = Elite, y = Outstate)) + 
  geom_boxplot()

# Histograms
library(gridExtra)
p1 <- ggplot(college, aes(x = Apps)) + 
  geom_histogram(bins = 50) + 
  labs(x = 'new applications')
p2 <- ggplot(college, aes(x = Enroll)) + 
  geom_histogram(bins = 45) + 
  labs(x = 'new enrollment')
p3 <- ggplot(college, aes(x = Expend)) + 
  geom_histogram(bins = 30) + 
  labs(x = 'Instructional expenditure per student')
p4 <- ggplot(college, aes(x = Outstate)) + 
  geom_histogram() + 
  labs(x = 'Out-of-state tuition')

grid.arrange(p1, p2, p3, p4, ncol = 2, nrow = 2)